go.mod

```diff
@@ -5,19 +5,21 @@ go 1.13
 require (
 	github.com/apparentlymart/go-dump v0.0.0-20190214190832-042adf3cf4a0 // indirect
 	github.com/aws/aws-sdk-go v1.25.3 // indirect
-	github.com/google/go-cmp v0.3.1 // indirect
-	github.com/gophercloud/gophercloud v0.17.0
+	github.com/gophercloud/gophercloud v0.18.0
 	github.com/gophercloud/utils v0.0.0-20210530213738-7c693d7efe47
 	github.com/hashicorp/go-retryablehttp v0.6.4
 	github.com/hashicorp/go-version v1.2.0
-	github.com/hashicorp/hcl v1.0.0 // indirect
 	github.com/hashicorp/hil v0.0.0-20190212132231-97b3a9cdfa93 // indirect
 	github.com/hashicorp/terraform-plugin-sdk v1.0.0
-	github.com/mattn/go-colorable v0.1.1 // indirect
-	github.com/mattn/go-isatty v0.0.7 // indirect
+	github.com/metal3-io/baremetal-operator v0.0.0-20211201170610-92ffa60c683d
```

Comment:
As far as I know, one of the reasons we've created `hardwareutils` module is not to import this. BMC package can be used from `hardwareutils/bmc` and raid functions might be copied from raid.go in here.

Reply:
I created a raid.go here to place raid functions from BMO.

Reply:
No, the idea of dropping the ironic module was that it should be fine to import all of the bmo module here.

Reply:
@zaneb I have updated this PR and imported all of the bmo module here. PTAL.